source/validate.cpp

```diff
@@ -215,6 +245,16 @@ spv_result_t ValidateBinaryUsingContextAndValidationState(
            << "Invalid SPIR-V header.";
   }
 
+
+  // Look for OpExtension instructions and register extensions.
+  // Diagnostics if any will be produced in the next pass (ProcessInstruction).
+  if (auto error = spvBinaryParse(&context, vstate, words, num_words,
+                                  setHeader, ProcessExtensions,
+                                  /* diagnostic = */ nullptr)) {
+    if (error != SPV_REQUESTED_TERMINATION)
```

Comment:
But then you never get a diagnostic in this early return case.  You should just let it fall through.